components/camel-http-base/src/main/java/org/apache/camel/http/base/HttpSendDynamicAware.java

```diff
@@ -100,6 +100,10 @@ public Processor createPreProcessor(Exchange exchange, DynamicAwareEntry entry)
             query = URISupport.createQueryString(new LinkedHashMap<>(entry.getLenientProperties()));
         }
 
+        if (query == null && exchange.getIn().getHeader(Exchange.HTTP_QUERY) != null) {
```

Comment:
Please use ObjectHelper

Reply:
Thanks for the suggestion, Used ObjectHelper to check the Header